Mlem/App/Utility/Extensions/Content Models/Interactable1Providing+Extensions.swift

```diff
@@ -180,6 +180,52 @@ extension Interactable1Providing {
         )
     }
     
+    func banActions() -> [any Action] {
+        let isModerator: Bool
+        if let myPerson = api.myPerson, let community = community_ {
+            isModerator = myPerson.moderates(communityId: community.id)
+        } else {
+            isModerator = false
+        }
+        var output: [any Action] = .init()
+        let showBoth = api.isAdmin && (bannedFromCommunity_ ?? false) != (creator_?.bannedFromInstance ?? false)
+        if api.isAdmin, !isModerator || showBoth {
+            output.append(banCreatorFromInstanceAction())
+        }
+        if isModerator || showBoth {
+            output.append(banCreatorFromCommunityAction())
+        }
```

Comment:
*nit*

This logic is correct but is very hard to read (it's a fairly complex operation--16 different input states -> 4 different output states). This is a little more verbose but makes the intended behavior much clearer:

```suggestion
        // admins should see separate 'ban' and 'unban' actions if ban statuses conflict; otherwise actions are grouped under a single entry
        let showBoth: Bool = api.isAdmin, (bannedFromCommunity_ ?? false) != (creator_?.bannedFromInstance ?? false)
        // moderators see community ban action by default, regardless of admin status
        if isModerator {
            if showBoth {
                output.append(banCreatorFromInstanceAction())
            }
            output.append(banCreatorFromCommunityAction())
        }
        else if api.isAdmin {
            output.append(banCreatorFromInstanceAction())
            if showBoth {
                output.append(banCreatorFromCommunityAction())
            }
        }
```

Reply:
Thanks, that's much cleaner :D